lib/api/core/auth/passportWrapper.js

```diff
@@ -0,0 +1,35 @@
+var
+  q = require('q'),
+  passport = require('passport');
+
+module.exports = function PassportWrapper (kuzzle) {
+
+  this.authenticate = function(request, strategy) {
+    var
+      deferred = q.defer();
+
+    try {
+      if (!passport._strategy(strategy)) {
+        return Promise.reject(new Error('Unknown authentication strategy "' + strategy + '"'));
```

Comment:
should be a ``xxxError`` (internal or badrequest, depending from where comes the ``strategy`` argument?), instead of a generic ``Error``